src/main/scala/chisel3/testing/HasTestingDirectory.scala

```diff
@@ -14,6 +14,30 @@ trait HasTestingDirectory {
   /** Return the directory where tests should be placed. */
   def getDirectory: Path
 
+  private def _getDirectory: Path = getDirectory
+
+  /** Factory that returns a new `HasTestingDirectory` which will put test files
+    * in a subdirectory of the parent `HasTestingDirectory`.
+    *
+    * For example, the object `bar` will have an output directory of `foo/bar/`.
+    * {{{
+    * import java.nio.file.Paths
+    *
+    * val foo = new HasTestingDirectory {
+    *   override def getDirectory = Paths.get("foo")
+    * }
+    *
+    * val bar = foo.getSubDir("bar")
+    * }}}
+    *
+    * @param subdirectory a subdirectory
+    */
+  def withSubdirectory(subdirectory: String): HasTestingDirectory = new HasTestingDirectory {
+
+    override def getDirectory = _getDirectory.resolve(subdirectory)
+
+  }
```

Comment:
Your `_getDirectory` pattern is fine, but the more common approach is to alias "this" (aka self-type with no type) the outer class:
```scala
trait HasTestingDirectory { outer =>
  def withSubdirectory(subdirectory: String): HasTestingDirectory = new HasTestingDirectory {
    override def getDirectory = outer.resolve(subdirectory)
  }
}

Reply:
Good suggestion. Fixed in https://github.com/chipsalliance/chisel/pull/4774/commits/b49bd29f63743e933dda1c0456e601ea61ccf87a.